pkg/db/db.go

```diff
@@ -181,10 +182,10 @@ func (c Client) UpdateMetadata(cacheDir string) error {
 
 	metadata, err := c.dbc.GetMetadata()
 	if err != nil {
-		return xerrors.Errorf("unable to get a metadata: %w", err)
+		return xerrors.Errorf("unable to get metadata: %w", err)
 	}
 
-	if err = c.metadata.Store(metadata); err != nil {
+	if err = c.dbc.StoreMetadata(metadata, filepath.Join(cacheDir, "db")); err != nil {
```

Comment:
We should remove the implementation of `metadata.Store`.
https://github.com/aquasecurity/trivy/blob/8e63667f44399b4eb9ed67391bcea2de84a4c94b/pkg/db/db.go#L214-L226

Also, I'm thinking if we move `Metadata` struct to `trivy-db`. Now, the metadata implementation exists in both trivy and trivy-db.

Reply:
So I tried removing the implementation of `metadata.Store` but I see that It's only used by tests but it's behaviour is needed for the corresponding `Get()` method to work as they have a dependency on https://github.com/aquasecurity/trivy/blob/dedfdcc67d631ad3ad22d19543d5b8645e2eecee/pkg/db/db.go#L195-L198

Changing the above is outside the scope of this PR. I propose we should leave a `TODO` in place to clean it up.

Reply:
You can change it as follows

```
b, err := json.Marshal(tc.metadata)
require.NoError(t, err)
err = afero.WriteFile(fs, "/cache/db/metadata.json", b, 0600)
require.NoError(t, err)
```

Of course, you should use a variable instead of `"/cache/db/metadata.json"`.

Actually, I feel we should move db.Metadata to trivy-db and `StoreMetadata` should be in the db.Metadata, not in db.Config. After that, we don't have to use `afero.WriteFile` directly. But as you said, we can do it in the next PR.

Reply:
Good point. I removed it here https://github.com/aquasecurity/trivy/pull/509/commits/a61d3ec879acfe15bf36b3f70b9a3d0d5fb06705